### PR TITLE
To support DV 215 - Currency to text

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx-Dataverse/issues/215.

### CODE REVIEW

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests/IRTests.cs [hunk 1]
```diff
@@ -203,5 +204,33 @@ public void BooleanOptionSetErrorTest(string expression)
             Assert.IsType<ErrorValue>(result);
             Assert.Contains("The value of option", ((ErrorValue)result).Errors.First().Message);
         }
+        
+        // Testing only currency to text and currency to decimal for the time being.
+        // Other coercion from currency should be added later.
+        [Theory]
+        [InlineData("Concatenate(First(MyTable).Currency, \"$\")", "CurrencyToText", "1$")]
+        [InlineData("Concatenate(\"$\", First(MyTable).Currency)", "CurrencyToText", "$1")]
+        [InlineData("Concatenate(First(MyTable).Currency * 100, \"%\")", "Decimal:w(FieldAccess(First:![Currency:$]", "100%")]
```
[the remark this answers is not in the document]
Comment (anderson-joyle) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests/IRTests.cs:213]: Good catch. It failed. Working on it!

File: src/tests/Microsoft.PowerFx.Core.Tests/PrimitiveValueConversionsTests.cs [hunk 2]
```diff
@@ -260,7 +260,6 @@ public override void Visit(ITypeVisitor vistor)
         [InlineData("D", "T", "UnaryOpKind.DateToTime")] // new
         [InlineData("D", "d", "UnaryOpKind.DateToDateTime")] // new
         [InlineData("n", "$", "UnaryOpKind.NumberToCurrency")] // new
-        [InlineData("$", "n", "UnaryOpKind.CurrencyToNumber")] // new
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/PrimitiveValueConversionsTests.cs:263]: >  [InlineData("$", "n", "UnaryOpKind.CurrencyToNumber")]  [](http://example.com/codeflow?start=6&length=57)

We should still test this conversion. I believe the result (coercionKindStr) now should be function:Float

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests/IRTests.cs [hunk 3]
```diff
@@ -203,5 +204,42 @@ public void BooleanOptionSetErrorTest(string expression)
...
+            Assert.Equal(expected, ((StringValue)result).Value);
+            Assert.Contains(ir, IRTranslator.Translate(check.Binding).ToString());
+
+            var check_float = engine.Check(expr, options: opt_float);
+            Assert.True(check_float.IsSuccess);
+
+            var result_float = check_float.GetEvaluator().Eval();
+            Assert.IsType<StringValue>(result_float);
+            Assert.Equal(expected, ((StringValue)result_float).Value);
+            Assert.Contains(ir_float, IRTranslator.Translate(check_float.Binding).ToString());
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests/IRTests.cs:242]: The two blocks are the same; consider combining them with something like

    foreach (var useFloat in new[] { false, true }) {
        var check = useFloat ? engine.Check(expr) : engine.Check(expr, opt_float);
        ... // al the rest is the same
    }

File: src/libraries/Microsoft.PowerFx.Core/IR/CoercionKind.cs [hunk 4]
```diff
@@ -107,6 +107,7 @@ internal enum CoercionKind
         CurrencyToText,
         TextToCurrency,
         CurrencyToBoolean,
-        BooleanToCurrency,
+        BooleanToCurrency,
+        CurrencyToDecimal
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/IR/CoercionKind.cs:111]: >CurrencyToDecimal [](http://example.com/codeflow?start=8&length=17)

Do we need a DecimalToCurrency conversion, if we need to send a number back to the table?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request closed without merging**